src/VisualStudio/Core/Def/Implementation/VirtualMemoryNotificationListener.cs

```diff
@@ -61,6 +68,17 @@ public int OnBroadcastMessage(uint msg, IntPtr wParam, IntPtr lParam)
                         }
 
                         _workspaceCacheService.FlushCaches();
+
+                        // turn off full solution analysis
+                        if (_workspace.Options.GetOption(RuntimeOptions.FullSolutionAnalysis))
+                        {
+                            _workspace.Services.GetService<IOptionService>().SetOptions(_workspace.Options.WithChangedOption(RuntimeOptions.FullSolutionAnalysis, false));
+
+                            // let user know full analysis is turned off due to memory concern
+                            // no close info bar action
+                            _workspace.Services.GetService<IErrorReportingService>().ShowErrorInfo(ServicesVSResources.FullSolutionAnalysisOff, () => { });
```

Comment:
> I thought about that before and choose not to do it since they will certainly OOM

But in your approach you then tell them how to re-enable things, which will then cause an OOM...

>  they can go ahead without restarting VS by set those options back on.

Or they could just click on the button :)

If they OOM, they OOM.  We told them they're in a low memory condition.  If they say "damnit... i still want to go ahead" why would we stop them?  Or, if we think we should stop them, then we shouldn't have a way for them to re-enable it in any way.

--

Also, the appraoch of telling someone "open tools|options, and then search for blah blah blah" is *exceptionally* user-unfriendly.  It's exactly waht people do not like.  "Why are you telling me how to do this when you could just go do it yourself?  This is so annoying for me to have to deal with!"